src/XTMF2.Web.Data/Converters/TypeConverter.cs

```diff
@@ -0,0 +1,48 @@
+//     Copyright 2017-2020 University of Toronto
+// 
+//     This file is part of XTMF2.
+// 
+//     XTMF2 is free software: you can redistribute it and/or modify
+//     it under the terms of the GNU General Public License as published by
+//     the Free Software Foundation, either version 3 of the License, or
+//     (at your option) any later version.
+// 
+//     XTMF2 is distributed in the hope that it will be useful,
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//     GNU General Public License for more details.
+// 
+//     You should have received a copy of the GNU General Public License
+//     along with XTMF2.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace XTMF2.Web.Data.Converters
+{
+    public class TypeConverter : JsonConverter<Type>
+    {
+        /// <summary>
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="typeToConvert"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public override Type Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            return Assembly.GetExecutingAssembly().GetType(reader.GetString());
```

Comment:
This might be a case for using [Type.GetType(String)](https://docs.microsoft.com/en-us/dotnet/api/system.type.gettype?view=netcore-3.1) in case the type is not defined in the currently executing assembly.